### PR TITLE
Improve partition boundaries

### PR DESCRIPTION
**Pull request by leerho**
This fixes a bug discovered in recent Druid testing.

Makes other improvements to the getPartionBoundaries code.

### CODE REVIEW

File: src/main/java/org/apache/datasketches/quantiles/ItemsSketchSortedView.java [hunk 1]
```diff
@@ -155,27 +159,68 @@ public GenericPartitionBoundaries<T> getPartitionBoundaries(final int numEqually
...
+    final long[] partNatRanks = new long[partArrLen];
+    final double[] partNormRanks = new double[partArrLen];
+    //adjust ends
+    int adjLen = svLen;
+    final boolean adjLow = quantiles[0] != minItem;
+    final boolean adjHigh = quantiles[svLen - 1] != maxItem;
+    adjLen += adjLow ? 1 : 0;
+    adjLen += adjHigh ? 1 : 0;
+
+    final T[] adjQuantiles;
```
Comment (jmalkin) [inline src/main/java/org/apache/datasketches/quantiles/ItemsSketchSortedView.java:191]: This is the start of a long code block with no indication of what it's doing. It'd be useful to have a little description of what's going on here.

File: src/test/java/org/apache/datasketches/sampling/EbppsSketchTest.java [hunk 2]
```diff
@@ -22,8 +22,10 @@
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertFalse;
 import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
 
 import java.util.ArrayList;
+import java.util.Arrays;
```
Comment (jmalkin) [inline src/test/java/org/apache/datasketches/sampling/EbppsSketchTest.java:28]: This is probably extraneous?

**Review by jmalkin — APPROVED** [review]
I don't know that I 100% get the tests in ItemsSketchPartitionBoundariesTest but I think they seemed fine.

Two minor comments (one a request for a comment, the other something that's probably not necessary). Nice to fix, not gonna block this on them.

File: src/test/java/org/apache/datasketches/sampling/EbppsSketchTest.java (hunk 2 above)
Comment (leerho) [reply src/test/java/org/apache/datasketches/sampling/EbppsSketchTest.java:28]: Yes, and so is: import static org.testng.Assert.fail;
I will fix.

File: src/main/java/org/apache/datasketches/quantiles/ItemsSketchSortedView.java (hunk 1 above)
Comment (leerho) [reply src/main/java/org/apache/datasketches/quantiles/ItemsSketchSortedView.java:191]: Good feedback.  Will do.

**Pull request merged**